### ISSUE TITLE
Removing a disk seems to completely bork the UI

### ISSUE THREAD

**Issue by jtilander** [rockstor/rockstor-core#1837]
Repro steps:

1. Setup a pool of 5 disks, RAID10.
2. Remove the first disk
3. Powercycle

Now when you go to the UI the following shows up with the following callstack:

```
            Traceback (most recent call last):
  File "/opt/rockstor/src/rockstor/rest_framework_custom/generic_view.py", line 41, in _handle_exception
    yield
  File "/opt/rockstor/src/rockstor/storageadmin/views/pool_balance.py", line 47, in get_queryset
    self._balance_status(pool)
  File "/opt/rockstor/eggs/Django-1.8.16-py2.7.egg/django/utils/decorators.py", line 145, in inner
    return func(*args, **kwargs)
  File "/opt/rockstor/src/rockstor/storageadmin/views/pool_balance.py", line 72, in _balance_status
    cur_status = balance_status(pool)
  File "/opt/rockstor/src/rockstor/fs/btrfs.py", line 985, in balance_status
    mnt_pt = mount_root(pool)
  File "/opt/rockstor/src/rockstor/fs/btrfs.py", line 252, in mount_root
    run_command(mnt_cmd)
  File "/opt/rockstor/src/rockstor/system/osi.py", line 115, in run_command
    raise CommandException(cmd, out, err, rc)
CommandException: Error running a command. cmd = /bin/mount /dev/disk/by-label/Homes /mnt2/Homes -o ,compress=lzo. rc = 32. stdout = ['']. stderr = ['mount: wrong fs type, bad option, bad superblock on /dev/sdd,', '       missing codepage or helper program, or other error', '', '       In some cases useful info is found in syslog - try', '       dmesg | tail or so.', '']

    
```

<img width="1120" alt="rockstor" src="https://user-images.githubusercontent.com/373005/31142294-b91e241c-a82e-11e7-89ba-3c01769f4ad4.png">

**Comment by phillxnet** [comment]
@jtilander Thanks for the report. What version of Rockstor are you running as we have made some recent improvements re UI representation of unmounted pools (volumes)? The default for btrfs, which we adhere to, is to not mount pools degraded. Which your pool would be if it had a disk removed. On our front we previously dealt poorly with this but as of #1763
https://forum.rockstor.com/t/dev-log-for-3-9-2/3532/8
we behave better and with the common follow up of employing ro,degraded mount options to retrieve data or achieve at least a ro mount we have another recent improvement in #1805.
https://forum.rockstor.com/t/dev-log-for-3-9-2/3532/26

However in your case there may be an additional wrinkle as your log output reports that a balance task may have been in effect (they resume after reboot), or triggered via a task for instance. It may be this wrinkle that is 'playing up' here. Could you provide the Rockstor version number and any other info re any balance tasks or any balance that may have been initiated via command line.

Thanks again for the report.

**Comment by jtilander** [comment]
Hi, sorry, I didn't mention that I was using this iso "Rockstor-3.9.1.iso" to install rockstor. (I'm new, so I was just downloading and trying your system out). 

I think you are correct, indeed there was a balance task running.

**Comment by phillxnet** [comment]
Cool; then I would say we have a new bug. It may take a little while to get around to this one as we are working towards the next stable release coming out shortly (sorry can't be more specific just yet). These are often accompanied by new iso releases but not always. In the mean time I would cancel your pending balance via the command line and if it is triggered via a scheduled task then disable that task. That should at least get you back to behaviour 'as is'. But until the pending changes to next stable are released you are welcome to subscribe to the testing channel updates to see how it goes. But do note it often breaks as it's pretty 'bleading edge'. Essentially Rockstor currently doesn't offer any repair assistance so you are back to command line tools for that side of things and a degraded pool is an example of where command line intervention is currently required (with the caveat of the above referenced closed issues in recent testing channel updates).

Thank again for chipping in with this one and please consider joining our forum (if you haven't already) as there are many there with knowledge of how to work around our current 'corner cases'.
https://forum.rockstor.com/

This issue should stay open until we deal more elegantly with an ongoing balance status enquiry in the contexts of a degraded pool on boot.

**Comment by phillxnet** [comment]
As a variant of this issue was observed during testing of pr:
"surface pool missing disk info in UI. Fixes #1897" #1902
I intend to make it my next focus.

Essentially there appears to be a balance status related error blocker the display of a degraded pool's details page. The pool overview was not found to be affected.

**Comment by phillxnet** [comment]
On further investigation it seems any unmounted pool, such as one that is degraded during boot and has no degraded mount option, that also has a balance record within the Rockstor UI, triggers the reported error within the UI for that pool's details page. The balance does not have to be 'in effect' as was previously thought. All other Web-UI pages are unaffected.

A similar record for Scrubs does not have the same effect. And adding ro,degraded as custom mount options to an affected pool will return the pool's details page to it's proper function; once the pool has successfully remounted.

It is proposed therefore that a more accurate title for this issue would be:

"unmounted pool with prior balance record has broken details page"
It is true that the indicated UI message is sticky but that is intended and a browser refresh in any other page will result in it disappearing. 

@jtilander do you fancy changing the title as suggested to reflect these findings?
